src/terminal/adapter/conGetSet.hpp

```diff
@@ -87,7 +88,8 @@ namespace Microsoft::Console::VirtualTerminal
         virtual BOOL PrivateEraseAll() = 0;
         virtual BOOL SetCursorStyle(const CursorType cursorType) = 0;
         virtual BOOL SetCursorColor(const COLORREF cursorColor) = 0;
-        virtual BOOL PrivateGetConsoleScreenBufferAttributes(_Out_ WORD* const pwAttributes) = 0;
+        virtual BOOL PrivateGetConsoleScreenBufferLegacyAttributes(_Out_ WORD* const pwAttributes) = 0;
+        virtual void PrivateGetConsoleScreenBufferAttributes(_Out_ TextAttribute& attributes) = 0;

```

Comment:
I think I'd prefer if this was `TextAttribute* const` instead of `TextAttribute&`. I'd rather ensure that whoever's using the `PrivateGetConsoleScreenBufferAttributes` call doesn't accidentally _modify_ the attributes of the active screen buffer.

Reply:
There is actually no danger of modifying the active screen buffer's stuff--the reference is passed in, meaning it refers to an object owned by a caller, and you can't "rebind" a C++ reference to refer to some other object. It's equivalent to a const pointer (like the other functions), except it cannot be null (thus I also got rid of the return value). This is why "modern C++" people might consider references superior, but a drawback that some people like myself don't like is that you can't tell at the callsite that you are passing a reference; it just looks like you are passing a normal argument by value, and hey wow, the callee mutated the argument?!

If this were my API, I would attempt to sidestep both issues: for all the getter functions that can only fail because the caller passed a null pointer, get rid of the pointer parameter, and just return the whatever-it-is by value (and thus make it no-fail).

All that said: this *isn't* my API, and I'd be happy to make this match all the others. (Even if it were my API, I'd modify the signatures in a separate change anyway.) I was just dashing something off to get the idea of my proposed change across. Thank you for looking at it and validating my approach! I'll go forward with prepping a "real" change.

Reply:
Okay that makes more sense. I think I wrote that comment a little too early in the morning and forgot how c++ worked :P.

You're right that maybe all these _should_ be references. I think this entire API surface was written when the style of the time within Windows was all pointers for everything, and when it comes to refactoring things that work well already, _ain't nobody got time for that_.